sherpa/astro/data.py

```diff
@@ -1761,27 +1761,87 @@ def get_specresp(self, filter=False):
 
         return newarf
 
-    # The energy bins can be grouped or ungrouped.  By default,
-    # they should be grouped if the data are grouped.  There are
-    # certain contexts (e.g., plotting) where we will retrieve the
-    # energy bins, and later filter the data; but filtering
-    # is automatically followed by grouping.  Grouping the data
-    # twice is an error.
     def _get_ebins(self, response_id=None, group=True):
+        """Return the low and high edges of the independent axis.
+
+        This method is badly named as it will return values in either
+        channel or energy units, depending on the units setting and
+        the associated response information. It also returns the
+        approximation of the mapping from channel space to energy -
+        that is the E_MIN and E_MAX columns from the RMF EBOUNDS block
+        rather than from the ENERG_LO and ENERG_HI columns from the
+        MATRIX block.
```

Comment:
But if no rmf is loaded, it takes the values from the arf (if present)

Reply:
This is one of those situations when properly describing the behavior ends up just repeating the code logic, so I didn't bother. If an ARF is present (but no RMF) then it uses the ENERG_LO/HI columns from the SPECRESP block. And technically the names MATRIX and SPECRESP aren't correct, as there are different variants. What I really wanted to say was that it's the ENERG_XX values and not E_XXX values (but, as you point out this is only meaningful for RMF responses).

Reply:
I've got around this by starting this sentence with the comment "If a RMF is loaded" (or words to that effect). This means that if we have an ARF-only analysis the reader is left clueless, but in that case we only have one way to specify energies.